service/kvm_config_framework.go

```diff
@@ -0,0 +1,195 @@
+
+		c.Address = config.Viper.GetString(config.Flag.Service.Kubernetes.Address)
+		c.InCluster = config.Viper.GetBool(config.Flag.Service.Kubernetes.InCluster)
+		c.TLS.CAFile = config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.CAFile)
+		c.TLS.CrtFile = config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.CrtFile)
+		c.TLS.KeyFile = config.Viper.GetString(config.Flag.Service.Kubernetes.TLS.KeyFile)
+
+		restConfig, err = k8srestconfig.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	clientSet, err := versioned.NewForConfig(restConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	k8sClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	k8sExtClient, err := apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var crdClient *k8scrdclient.CRDClient
+	{
+		c := k8scrdclient.DefaultConfig()
+
+		c.K8sExtClient = k8sExtClient
+		c.Logger = config.Logger
+
+		crdClient, err = k8scrdclient.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var certsSearcher certs.Interface
+	{
+		c := certs.DefaultConfig()
+
+		c.K8sClient = k8sClient
+		c.Logger = config.Logger
+
+		certsSearcher, err = certs.NewSearcher(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var randomkeysSearcher randomkeys.Interface
+	{
+		keyConfig := randomkeys.DefaultConfig()
+		keyConfig.K8sClient = k8sClient
+		keyConfig.Logger = config.Logger
+		randomkeysSearcher, err = randomkeys.NewSearcher(keyConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourcesV2 []framework.Resource
+	{
+		c := defaultKVMConfigResourcesV2Config()
+
+		c.CertsSearcher = certsSearcher
+		c.K8sClient = k8sClient
+		c.Logger = config.Logger
+		c.RandomkeysSearcher = randomkeysSearcher
+
+		c.Name = config.Name
+
+		resourcesV2, err = newKVMConfigResourcesV2(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resourcesV3 []framework.Resource
+	{
+		c := defaultKVMConfigResourcesV3Config()
+
+		c.CertsSearcher = certsSearcher
+		c.K8sClient = k8sClient
+		c.Logger = config.Logger
+		c.RandomkeysSearcher = randomkeysSearcher
+
+		c.Name = config.Name
+
+		resourcesV3, err = newKVMConfigResourcesV3(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	// We provide a map of resource lists keyed by the version bundle version
+	// to the resource router.
+	versionedResources := map[string][]framework.Resource{
+		"1.1.0": resourcesV3,
+		"1.0.0": resourcesV2,
+		"0.1.0": resourcesV2,
+		"":      resourcesV2,
+	}
+
+	var newInformer *informer.Informer
+	{
+		c := informer.DefaultConfig()
+
+		c.Watcher = clientSet.ProviderV1alpha1().KVMConfigs("")
+
+		newInformer, err = informer.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	// TODO route initCtx func together with resources.
```

Comment:
It would be nice to route initCtxFunc together with []framework.Resource. That way we wouldn't have to do `if`s on versionBundleVersion.
This is next steps. Requires changes in operatorkit.

Reply:
How do we handle unknown version bundles or exceptions then?